src/pages/Portfolio/ImpactDashboard/YourDonations.vue

```diff
@@ -0,0 +1,277 @@
+import AsyncPortfolioSection from './AsyncPortfolioSection';
+import KvGrid from '~/@kiva/kv-components/vue/KvGrid';
+import KvLightbox from '~/@kiva/kv-components/vue/KvLightbox';
+import KvButton from '~/@kiva/kv-components/vue/KvButton';
+import KvLoadingPlaceholder from '~/@kiva/kv-components/vue/KvLoadingPlaceholder';
+import KvMaterialIcon from '~/@kiva/kv-components/vue/KvMaterialIcon';
+
+export default {
+	name: 'YourDonations',
+	inject: ['apollo'],
+	components: {
+		AsyncPortfolioSection,
+		KvLoadingPlaceholder,
+		KvMaterialIcon,
+		KvLightbox,
+		KvButton,
+		KvGrid,
+	},
+	data() {
+		return {
+			loading: true,
+			loadingPromise: null,
+			averageDonation: 0,
+			donationLightboxVisible: false,
+			currentDate: new Date(),
+			currentYearDonations: 0,
+			lastYearDonations: 0,
+			lastYear: null,
+			currentYear: null,
+			taxReceiptUrl: '',
+			totalDonations: 0,
+			limit: 100,
+			mdiArrowRight
+		};
+	},
+	methods: {
+		fetchAsyncData() {
+			const startDate = (new Date(this.currentDate.getFullYear() - 1, 0, 1)).toISOString();
+			const endDate = (new Date(this.currentDate.getFullYear(), 11, 31)).toISOString();
+
+			this.lastYear = this.currentDate.getFullYear() - 1;
+			this.currentYear = this.currentDate.getFullYear();
+			this.taxReceiptUrl = `https://${this.$appConfig.host}/portfolio/donations/print?year=${this.lastYear}`;
+
+			const transactionFilters = {
+				category: 'donation',
+				startDate,
+				endDate,
+			};
+
+			if (this.loading && !this.loadingPromise) {
+				this.loadingPromise = this.apollo.query({
+					query: gql`query yourDonations ($filters: TransactionSearchFilters, $limit: Int) {
+						my {
+							id
+							transactions(filter: $filters, limit: $limit) {
+								totalCount
+								values {
+									createTime
+									amount
+								}
+							}
+							userStats {
+								amount_donated
+								amount_of_loans
+							}
+						}
+					}`,
+					variables: {
+						filters: transactionFilters,
+						limit: this.limit,
+					},
+				}).then(({ data }) => {
+					this.loading = false;
+					const currentYear = this.currentDate.getFullYear();
+					const transactions = data?.my?.transactions?.values ?? [];
+					const amountOfLoans = numeral(data?.my?.userStats?.amount_of_loans ?? 0);
+					const donation = numeral(data?.my?.userStats?.amount_donated ?? 0);
+					if (amountOfLoans.value() > 0) {
+						donation.multiply(1 / amountOfLoans.value()).multiply(25);
+					}
+					this.averageDonation = donation.format('$0,0[.]00');
+
+					this.totalDonations = data?.my?.transactions?.totalCount ?? 0;
+					this.currentYearDonations = this.filterAndSumDonations(transactions, currentYear);
+					this.lastYearDonations = this.filterAndSumDonations(transactions, currentYear - 1);
+
+					if (this.totalDonations > this.limit) {
+						let offset = this.limit;
+						while (offset < this.totalDonations) {
+							this.getRemainingDonations(transactionFilters, offset)
```

Comment:
The way this works right now, the entire while loop will execute synchronously, kicking off a bunch of parallel requests for the rest of the results, and then complete before any of the results can come in. The `finally()` callback will be called before the rest of those requests complete.

You could maybe build an array of requests based on the totalCount and then return Promise.all(requests)